Dockerfile

```diff
@@ -7,15 +7,54 @@ RUN apt-get update && apt-get install -y \
   mediainfo \
   unzip \
   && rm -rf /var/lib/apt/lists/*
+RUN php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
+RUN php composer-setup.php --install-dir=/bin --filename=composer --version=1.10.16
+RUN php -r "unlink('composer-setup.php');"
+
+# Set Timezone
+RUN echo "date.timezone = Europe/London" > /usr/local/etc/php/conf.d/timezone_set.ini
+
+# Remove the memory limit for the CLI only.
+RUN echo 'memory_limit = -1' > /usr/local/etc/php/php-cli.ini
+
+# Remove the vanilla Drupal ready to install a dev version
+RUN rm -rf ..?* .[!.]* *
+
+# Install Drupal 8.x Dev
+RUN composer create-project drupal-composer/drupal-project:8.x-dev . --stability dev --no-interaction
+
+# Update autoloads
+RUN composer dump-autoload --optimize
 
+# Install custom modules and run PHPUnit
+WORKDIR /opt/drupal/web
+COPY phpunit.xml core/phpunit.xml
+COPY modules/custom modules/custom
+RUN ../vendor/bin/phpunit -c core --testsuite unit --debug --verbose
+
+###########################################################################################
+# Create runtime image
+###########################################################################################
+
+FROM drupal:8.9-apache
+
+# Install Composer and it's dependencies
+RUN apt-get update && apt-get install -y \
+  curl \
+  git-core \
+  mediainfo \
+  unzip \
+  && rm -rf /var/lib/apt/lists/*
```

Comment:
🙌 loving the multi-stage Dockerfile ❤️ 

We can dry up the commonality between test and production builds using an interim step, e.g. `base`:

```Dockerfile
FROM drupal:8.9.2-apache as base
RUN apt-get update && apt-get install ...

FROM base as test
RUN rm -rf ..?* .[!.]* *
RUN composer create-project drupal-composer/drupal-project:8.x-dev . --stability dev --no-interaction

FROM base as production
RUN composer dump-autoload --optimize
```

Sorry if I'm double-commenting, but I thought I'd added this and can't see it...

Reply:
Updated the `Dockerfile` and pinned to `8.9.6`